src/sentry/static/sentry/app/components/events/interfaces/debugMeta-v2/index.tsx

```diff
@@ -0,0 +1,433 @@
+
+  filterImages() {
+    // skip null values indicating invalid debug images
+    const debugImages = this.getDebugImages();
+    const filteredImages = debugImages.filter(image => this.filterImage(image));
+
+    this.setState({filteredImages}, this.updateGrid);
+  }
+
+  getListHeight() {
+    const {panelTableHeight} = this.state;
+
+    if (!panelTableHeight || panelTableHeight > PANEL_MAX_HEIGHT) {
+      return PANEL_MAX_HEIGHT;
+    }
+
+    return panelTableHeight;
+  }
+
+  handleChangeFilter = (value = '') => {
+    DebugMetaActions.updateFilter(value);
+  };
+
+  handleOpenImageDetailsModal = (
+    image: Image,
+    imageAddress: React.ReactElement | null,
+    fileName?: string
+  ) => {
+    const {organization, projectId} = this.props;
+    return openModal(
+      modalProps => (
+        <DebugImageDetails
+          {...modalProps}
+          image={image}
+          title={fileName}
+          organization={organization}
+          projectId={projectId}
+          imageAddress={imageAddress}
+        />
+      ),
+      {
+        modalCss,
+      }
+    );
+  };
+
+  renderRow = ({index, key, parent, style}: ListRowProps) => {
+    const {filteredImages} = this.state;
+
+    return (
+      <CellMeasurer
+        cache={cache}
+        columnIndex={0}
+        key={key}
+        parent={parent}
+        rowIndex={index}
+      >
+        <DebugImage
+          style={style}
+          image={filteredImages[index]}
+          onOpenImageDetailsModal={this.handleOpenImageDetailsModal}
+        />
+      </CellMeasurer>
+    );
+  };
+
+  renderList() {
+    const {filteredImages: images, panelTableHeight} = this.state;
+
+    if (!panelTableHeight) {
+      return images.map(image => (
+        <DebugImage
+          key={image.debug_id}
+          image={image}
+          onOpenImageDetailsModal={this.handleOpenImageDetailsModal}
+        />
+      ));
+    }
+
+    return (
+      <AutoSizer disableHeight onResize={this.updateGrid}>
+        {({width}) => (
+          <StyledList
+            ref={(el: List | null) => {
+              this.listRef = el;
+            }}
+            deferredMeasurementCache={cache}
+            height={this.getListHeight()}
+            overscanRowCount={5}
+            rowCount={images.length}
+            rowHeight={cache.rowHeight}
+            rowRenderer={this.renderRow}
+            onScrollbarPresenceChange={this.setScrollbarSize}
+            width={width}
+            isScrolling={false}
+          />
+        )}
+      </AutoSizer>
+    );
+  }
+
+  render() {
+    const {filter, filteredImages: images, scrollbarSize} = this.state;
+
+    return (
+      <StyledEventDataSection
+        type="images-loaded"
+        title={
+          <TitleWrapper>
+            <GuideAnchor target="images-loaded" position="bottom">
+              <Title>{t('Images Loaded')}</Title>
+            </GuideAnchor>
+            <QuestionTooltip size="xs" position="top" title="This is a description" />
```

Comment:
@jan-auer could you please add a description here?

Reply:
Added.